codelabs/codelab-02/codelab2README.md

```diff
@@ -0,0 +1,507 @@
+
+You can see the full list of services from the "Services" dropdown in the top-left corner:
+
+![Services Tab](../../../media/codelabs/codelab-02/services.png)
+
+You can also use either search bar to locate services.
+
+To see which region you are currently in, look for the region tab in the top-right corner:
+
+![Regions Tab](../../../media/codelabs/codelab-02/regions.png)
+
+Keep in mind that we will be using us-east-1, which is in North Virginia.
+
+As mentioned above, we will be using S3 in this codelab, so I would recommend that you go ahead and take a peek at S3 via the AWS Console. Come back throughout the codelab and look at how things change as you create buckets, upload files, etc.
+
+![S3 Splash Page](../../../media/codelabs/codelab-02/s3.png)
+
+#### AWS CLI
+
+##### IAM User
+
+To work with the CLI, you'll need to first create an IAM user via the AWS Console. Open the "IAM" service and navigate to "Users" (tab on left) > "Add user".
+
+1. Pick a name for your user, and check "Programmatic Access". You won't need to sign in to the AWS console with your user.
+
+![Create User](../../../media/codelabs/codelab-02/create-user.png)
+
+2. You need to specify what permissions your new user has. To do this, we will create an IAM group for admin users. Give this group a name and select the "AdministratorAccess" policy.
+
+![Admin Permissions](../../../media/codelabs/codelab-02/admin-group.png)
+
+3. Click through to the "Complete" page where AWS will provide you with the access key ID and secret access key. Keep the latter safe, perhaps saving them in a text file on your machine, because AWS will not show it to you again. These keys are used to sign programmatic requests that you make to AWS.
+
+##### Testing the CLI
+
+If you haven't already done so, set up your local development environment. Follow the instructions here: [Environment Setup](https://github.com/UMD-CS-STICs/389Lspring18/blob/master/env.md). Upon entering the final command, you will be prompted to give the keys you just obtained. Set us-east-1 and json as the default region and output respectively.
+
+$ aws configure 
+AWS Access Key ID [None]: AKIAIOSFODNN7EXAMPLE 
+AWS Secret Access Key [None]: wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY 
+Default region name [None]: us-east-1 
+Default output format [None]: json
+
+Now test your credentials with the following command.
+
+$ aws its get-caller-identity
+
+Good to go. We'll use the CLI more below.
+
+##### Root User
+
+To clarify the difference between the user you have been using to log in to the AWS Console and the IAM user you just created:
+
+The credentials you've been using to log in with are the root user credentials. Normally, you avoid using the root account and instead federate access out to IAM users so that you can audit their usage. However, in this class, we will continue to use the root account for simplicity.
```

Comment:
Maybe rephrase:

Essentially, you have two users for your account: the **root user**, which you use to sign-in to the AWS console, and the **IAM user** (which has an Account ID and Secret Access Key), which you use to programmatically access AWS.

We can't programmatically access the API with the root user, because it does not have associated access keys. In general, you tend to avoid using the root account on teams since it is hard to audit. Preferably all team members have an IAM user they can access, with a limited set of permissions.

However, in this class, we will continue to use the root account for simplicity.

Reply:
Though we could also completely dump using the root user, and instead create an IAM user which also has "AWS Management Console access". I think this might actually be the better way, what do you think @iaLockwood?